### PR TITLE
docs(Icon): more detailed migration guide

### PR DESCRIPTION
**Pull request by asudoh**
Refs #3771.

#### Changelog

**New**

- Migration guide specific for `carbon-components-react`'s `<Icon>` component.

### CODE REVIEW

**Review by emyarod — APPROVED** [review]
looks good to me!

**Review by jnm2377 — APPROVED** [review]

File: packages/react/src/components/Icon/migrate-to-7.x.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+| Prop   | v9                      | v10                 |
+| ------ | ----------------------- | ------------------- |
+| `name` | Points to the icon name | Removed (See below) |
+
+## Notes for `v10`
+
+`<Icon>` component is for using `carbon-icons` library which hosts older icons,
+and our React components in `v10` uses `@carbon/icons-react` library which hosts
+newer icons introduced in `v10` and works right as React components.
+
```
Comment (joshblack) [inline packages/react/src/components/Icon/migrate-to-7.x.md]: Maybe something like:

> The `<Icon>` component uses the `carbon-icons` library, which hosts a set of icons built for v9 of the Carbon Design System. In v10, we introduced `@carbon/icons-react` which offers an updated set of icons built for consumption as React components. However, we will still support both `carbon-icons` and `@carbon/icons-react` through v11.

File: packages/react/src/components/Icon/migrate-to-7.x.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+You can still use `<Icon>`, but `name` prop of `<Icon>` has been removed in
```
Comment (joshblack) [inline packages/react/src/components/Icon/migrate-to-7.x.md]:
```suggestion
While you can still use the `<Icon>` component, the `name` prop has been removed in
```

File: packages/react/src/components/Icon/migrate-to-7.x.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+`v10`, given the name-lookup approach loses the ability of pick and choose icons
+to be included in application's build. The alternative is `icon` prop, which
+takes a
+[data structure](https://github.com/carbon-design-system/carbon/blob/v10.3.0/packages/react/src/components/Icon/Icon.js#L189-L194)
+seen in
+[named exports of `carbon-icons` library](https://unpkg.com/browse/carbon-icons@7.0.1/dist/carbon-icons-list.js).
```
Comment (joshblack) [inline packages/react/src/components/Icon/migrate-to-7.x.md]: Maybe something like:

> ...the `name` prop has been removed in `v10`. We removed the `name` prop in preference of the `icon` prop which will allow you to tree-shake icons in your build. You can use the `icon` prop by directly importing icons from `carbon-icons` and supplying them as the `icon` prop. For example:

```js
import { Icons } from 'carbon-components-react';
import { iconAdd } from 'carbon-icons';

<Icon icon={iconAdd} />
```

This `icon` prop is a data structure that we use to represent an icon in code. You can pass in your own icons if they follow this data structure. For more information, you can view the [`icon` prop type](https://github.com/carbon-design-system/carbon/blob/v10.3.0/packages/react/src/components/Icon/Icon.js#L189-L194) in addition to the exports from [`carbon-icons`](https://unpkg.com/browse/carbon-icons@7.0.7/dist/carbon-icons-list.js).

File: packages/react/src/components/Icon/migrate-to-7.x.md (hunk 1 above)
Comment (asudoh) [reply packages/react/src/components/Icon/migrate-to-7.x.md]: Makes sense thanks - Updated.

File: packages/react/src/components/Icon/migrate-to-7.x.md (hunk 3 above)
Comment (asudoh) [reply packages/react/src/components/Icon/migrate-to-7.x.md]: Makes sense thanks - Updated.

**Review by joshblack — APPROVED** [review]

**Pull request merged**